File: base/--build/tmp/kapt3/stubs/debug/com/shopify/shopifyapp/ordersection/activities/OrderList.java
package com.baytrendy.shopifyapp.ordersection.activities;

import java.lang.System;

@kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000Z\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000e\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010!\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0003\u0018\u00002\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0002J\u0010\u0010\u001c\u001a\u00020\u001d2\u0006\u0010\u001e\u001a\u00020\u001fH\u0002J\u0012\u0010 \u001a\u00020\u001d2\b\u0010!\u001a\u0004\u0018\u00010\"H\u0014J\u0010\u0010#\u001a\u00020\u001d2\u0006\u0010$\u001a\u00020\u0014H\u0002R\u001e\u0010\u0003\u001a\u00020\u00048\u0006@\u0006X\u0087.\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\u0005\u0010\u0006\"\u0004\b\u0007\u0010\bR\u0010\u0010\t\u001a\u0004\u0018\u00010\nX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u001e\u0010\u000b\u001a\u00020\f8\u0006@\u0006X\u0087.\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\r\u0010\u000e\"\u0004\b\u000f\u0010\u0010R\u0010\u0010\u0011\u001a\u0004\u0018\u00010\u0012X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\u0013\u001a\u0004\u0018\u00010\u0014X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\u0015\u001a\u0004\u0018\u00010\u0016X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0016\u0010\u0017\u001a\n\u0012\u0004\u0012\u00020\u0019\u0018\u00010\u0018X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u001a\u001a\u00020\u001bX\u0082\u0004\u00a2\u0006\u0002\n\u0000\u00a8\u0006%"}, d2 = {"Lcom/shopify/shopifyapp/ordersection/activities/OrderList;", "Lcom/shopify/shopifyapp/basesection/activities/BaseActivity;", "()V", "adapter", "Lcom/shopify/shopifyapp/ordersection/adapters/OrderListAdapter;", "getAdapter", "()Lcom/shopify/shopifyapp/ordersection/adapters/OrderListAdapter;", "setAdapter", "(Lcom/shopify/shopifyapp/ordersection/adapters/OrderListAdapter;)V", "binding", "Lcom/shopify/shopifyapp/base/databinding/MOrderlistBinding;", "factory", "Lcom/shopify/shopifyapp/utils/ViewModelFactory;", "getFactory", "()Lcom/shopify/shopifyapp/utils/ViewModelFactory;", "setFactory", "(Lcom/shopify/shopifyapp/utils/ViewModelFactory;)V", "model", "Lcom/shopify/shopifyapp/ordersection/viewmodels/OrderListViewModel;", "ordercursor", "", "orderlist", "Landroidx/recyclerview/widget/RecyclerView;", "orders", "", "Lcom/shopify/buy3/Storefront$OrderEdge;", "recyclerViewOnScrollListener", "Landroidx/recyclerview/widget/RecyclerView$OnScrollListener;", "consumeResponse", "", "response", "Lcom/shopify/buy3/Storefront$OrderConnection;", "onCreate", "savedInstanceState", "Landroid/os/Bundle;", "showToast", "msg", "base_debug"})
public final class OrderList extends com.baytrendy.shopifyapp.basesection.activities.BaseActivity {
    private com.baytrendy.shopifyapp.databinding.MOrderlistBinding binding;
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Inject()
    public com.baytrendy.shopifyapp.utils.ViewModelFactory factory;
    private androidx.recyclerview.widget.RecyclerView orderlist;
    private com.baytrendy.shopifyapp.ordersection.viewmodels.OrderListViewModel model;
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Inject()
    public com.baytrendy.shopifyapp.ordersection.adapters.OrderListAdapter adapter;
    private java.util.List<com.shopify.buy3.Storefront.OrderEdge> orders;
    private java.lang.String ordercursor;
    private final androidx.recyclerview.widget.RecyclerView.OnScrollListener recyclerViewOnScrollListener = null;
    private java.util.HashMap _$_findViewCache;
    
    @org.jetbrains.annotations.NotNull()
    public final com.baytrendy.shopifyapp.utils.ViewModelFactory getFactory() {
        return null;
    }
    
    public final void setFactory(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.utils.ViewModelFactory p0) {
    }
    
    @org.jetbrains.annotations.NotNull()
    public final com.baytrendy.shopifyapp.ordersection.adapters.OrderListAdapter getAdapter() {
        return null;
    }
    
    public final void setAdapter(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.ordersection.adapters.OrderListAdapter p0) {
    }
    
    @java.lang.Override()
    protected void onCreate(@org.jetbrains.annotations.Nullable()
    android.os.Bundle savedInstanceState) {
    }
    
    private final void showToast(java.lang.String msg) {
    }
    
    private final void consumeResponse(com.shopify.buy3.Storefront.OrderConnection response) {
    }
    
    public OrderList() {
        super();
    }
}

File: base/src/main/java/com/baytrendy/shopifyapp/ordersection/activities/OrderList.kt
package com.baytrendy.shopifyapp.ordersection.activities

import android.os.Bundle
import android.util.Log
import android.view.ViewGroup
import android.widget.Toast

import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.shopify.buy3.Storefront
import com.baytrendy.shopifyapp.MyApplication
import com.baytrendy.shopifyapp.R
import com.baytrendy.shopifyapp.databinding.MOrderlistBinding
import com.baytrendy.shopifyapp.basesection.activities.NewBaseActivity
import com.baytrendy.shopifyapp.ordersection.adapters.OrderListAdapter
import com.baytrendy.shopifyapp.ordersection.viewmodels.OrderListViewModel
import com.baytrendy.shopifyapp.utils.ViewModelFactory

import javax.inject.Inject

class OrderList : NewBaseActivity() {
    private var binding: MOrderlistBinding? = null

    @Inject
    lateinit var factory: ViewModelFactory
    private var orderlist: RecyclerView? = null
    private var model: OrderListViewModel? = null

    @Inject
    lateinit var adapter: OrderListAdapter
    private var orders: MutableList<Storefront.OrderEdge>? = null
    private var ordercursor: String? = null
    private val recyclerViewOnScrollListener = object : RecyclerView.OnScrollListener() {
        override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
            super.onScrollStateChanged(recyclerView, newState)
        }

        override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
            super.onScrolled(recyclerView, dx, dy)
            val visibleItemCount = recyclerView.layoutManager!!.childCount
            val totalItemCount = recyclerView.layoutManager!!.itemCount
            val firstVisibleItemPosition = (recyclerView.layoutManager as LinearLayoutManager).findFirstVisibleItemPosition()
            if (!recyclerView.canScrollVertically(1)) {
                if (visibleItemCount + firstVisibleItemPosition >= totalItemCount && firstVisibleItemPosition > 0
                        && totalItemCount >= orders!!.size) {
                    model!!.cursor = ordercursor!!
                }
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val group = findViewById<ViewGroup>(R.id.container)
        binding = DataBindingUtil.inflate(layoutInflater, R.layout.m_orderlist, group, true)
        orderlist = setLayout(binding!!.orderlist, "vertical")
        orderlist!!.addOnScrollListener(recyclerViewOnScrollListener)
        showTittle(resources.getString(R.string.myorders))
        showBackButton()
        (application as MyApplication).mageNativeAppComponent!!.doOrderListInjection(this)
        model = ViewModelProviders.of(this, factory).get(OrderListViewModel::class.java)
        model!!.context = this
        model!!.errorResponse.observe(this, Observer<String> { this.showToast(it) })
        model!!.getResponse_().observe(this, Observer<Storefront.OrderConnection> { this.consumeResponse(it) })
    }

    private fun showToast(msg: String) {
        Toast.makeText(this, msg, Toast.LENGTH_LONG).show()
    }

    private fun consumeResponse(response: Storefront.OrderConnection) {
        try {
            if (response.edges.size > 0) {
                if (this.orders == null) {
                    this.orders = response.edges
                    adapter!!.setData(orders as MutableList<Storefront.OrderEdge>?, model)
                    orderlist!!.adapter = adapter
                } else {
                    this.orders!!.addAll(response.edges)
                    adapter!!.notifyDataSetChanged()
                }
                ordercursor = this.orders!![this.orders!!.size - 1].cursor
                Log.i("MageNative", "Cursor : " + ordercursor!!)
            } else {
                showToast(resources.getString(R.string.noorders))
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }
}
